src/ThemeSupport/ThemeSupport.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace WPGatsby\ThemeSupport;
+
+/**
+ * Modifies the schema
+ */
+class ThemeSupport {
+	/**
+	 *
+	 */
+	function __construct() {
+        add_action( 'init', [ $this, 'registerGatsbyMenuLocations' ] );
+    }
+    
+    function registerGatsbyMenuLocations() {
```

Comment:
I'm not sure I fully support this move to register default locations. Gatsby should work with any location folks have or want to have in the future. Is there a technical reason we _need_ this other than ` it was hella annoying switching themes`

I think if there's a legit technical reason, like the source plugin absolutely won't work without these locations, I suppose we can go this route, but I don't love adding this if we don't have to. Folks should be able to use existing locations, register others named whatever they want to name them, and Gatsby should be able to work with them.

Reply:
It does work with existing ones, the annoying part was not having consistent locations. Folks can register or deregister whatever they want and it'll keep working. This is just so you can enable nude theme or some other frontend-less theme and still have consistent theme locations.

Reply:
Any location will work with this logic. I tested it on the 2020 theme as well as with these custom registered locations

Reply:
So this change is adjacent to the issue, but I think we need some default locations regardless. Especially if we don't want folks to have to write PHP necessarily. I think long term we should have a UI where you can remove/change the defaults or add new ones

Reply:
If we don't _need_ these, I'd prefer we leave them out. This feels more like a documentation thing to me. We can provide this as a snippet folks can use if they want consistent locations across active themes, but I don't think we need to enforce this. I think it will actually lead to confusion where folks might think the _only_ way to use menus with Gatsby is with one of these locations.

Reply:
One more thought about it: right now if you use 2020 locations, changing your theme will break your Gatsby build. I think locations used by Gatsby shouldn't dissapear unless WPGatsby is disabled. Otherwise it can lead to unexpected build failures

Reply:
Same for any other theme locations. I think folks will do that without realizing it, so this is a build consistency improvement

Reply:
I'd rather document why other locations not registered by WPGatsby might be flaky

Reply:
To me it comes back to reducing friction as much as possible. This is a huge QOL improvement for folks using this stack. Then when we add a UI to change the WPGatsby locations it will be feature complete

Reply:
Changing a theme will likely break a LOT of things. This is common when changing themes with monolithic WordPress too. 

If you change themes, you often expect to rebuild a lot of things. 

If you feel strongly about it, I think it should be disabled by default and can be enabled via a setting. 

The developer enabling WPGatsby isn't the only person working on a site. If one of our editors at one of the newspapers all of a sudden saw a bunch of new menu locations it would stress them out, lol. Then an hour would be wasted with them talking to our PM, PM asking us questions, etc, etc. 

Then for the sites that have 4 footer menus, or different menus for different templates, etc are going to have to register code themselves anyway. 

This feels like user land code, not something we _need_ to provide from WPGatsby.

Reply:
Fair enough, but consider this: When you're using monoWP and install something like Avada, Avada has an internal contract where it can garuntee the menu's it adds can possibly exist. `avada-custom-menu-1` will always be possible. When you go headless you can't make that garuntee anymore. Since Gatsby is now acting as the theme layer, it makes sense that it should also be able to provide some kind of garuntee about what data can be available. The only thing that should be able to break the site in the majority of cases is disabling WPGatsby since WPGatsby is the theme orchestrator here.

Reply:
Also that's why I added `[Gatsby]`, so folks know where it came from. If this was a plugin option I would tell everyone to enable it and use it, so I think it should be opt-out not opt-in. I would literally write it in the docs as a best practise to make sure your build doesn't break :p What if we update it to say `[Added by WPGatsby]` instead of `[Gatsby]`?

Reply:
@jasonbahl I shrunk the list of locations down to 2 (header and footer), made the added locations filterable, and added an admin setting to disable them. Thanks for the suggestions!

Reply:
Oh I also change it to say `[Added by WPGatsby]` instead of `[Gatsby]` so it's clearer that these are added by the plugin